src/midiEditorCommands.cpp

```diff
@@ -26,6 +27,29 @@
 using namespace std;
 using namespace fmt::literals;
 
+// returns the pulses per quarter note of the first midi source in the item.
+int getItemPPQ(MediaItem* item) {
+	const int defaultPPQ = 960;
+	static MediaItem* cachedItem;
```

Comment:
This caches the media item for as long as REAPER is running. Would it be possible to refresh the cached item for as long as the event list is open?
Not sure what `kbd_OnMidiList` does exactly, but may be that can be used?

Reply:
It caches PPQ for the item until it is called with a different item. Do you know if it is possible to change PPQ after an item is created? It isn't possible to change it  in the item source properties.

Reply:
I think you're right that the ppq will stay always the same then. The option that allows you to change ppq says new items explicitly.